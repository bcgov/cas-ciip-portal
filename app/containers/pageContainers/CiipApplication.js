import React, {Component} from 'react';
import {Jumbotron} from 'react-bootstrap';
import {graphql} from 'react-relay';
import DefaultLayout from '../../layouts/default-layout';
import ApplicationWizard from '../Applications/ApplicationWizard';

class CiipApplication extends Component {
  static getInitialProps = () => {
    return {variables: {formCondition: {rowId: -1}}};
  };

  static query = graphql`
    query CiipApplicationQuery($formCondition: FormJsonCondition!) {
      query {
        ...ApplicationWizard_query @arguments(formCondition: $formCondition)
      }
    }
  `;

  render() {
    const {query} = this.props;
    return (
      <DefaultLayout>
        <ApplicationWizard query={query} />
      </DefaultLayout>
    );
  }
}

export default CiipApplication;
