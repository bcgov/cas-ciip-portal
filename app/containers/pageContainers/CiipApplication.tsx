import React, {Component} from 'react';
import {graphql} from 'react-relay';
import DefaultLayout from '../../layouts/default-layout';
import ApplicationWizard from '../Applications/ApplicationWizard';

class CiipApplication extends Component {
  static query = graphql`
    query CiipApplicationQuery($formId: ID!, $applicationId: ID!) {
      query {
        ...ApplicationWizard_query
          @arguments(formId: $formId, applicationId: $applicationId)
      }
    }
  `;

  static getInitialProps = () => ({
    variables: {
      formId: ''
    }
  });

  render() {
    const {query} = this.props;
    return (
      <DefaultLayout>
        <ApplicationWizard query={query} />
      </DefaultLayout>
    );
  }
}

export default CiipApplication;
