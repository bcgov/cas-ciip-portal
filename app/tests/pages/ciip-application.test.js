import React from 'react';
import {shallow} from 'enzyme';
import CiipApplication from '../../containers/pageContainers/CiipApplication';

const query = {
  allFormJsons: {
    edges: [{node: {id: 'form-1'}}]
  }
};

// It matches the last accepted Snapshot
it('It matches the last accepted Snapshot', () => {
  const wrapper = shallow(<CiipApplication />);
  expect(wrapper).toMatchSnapshot();
});

it('It passes a query to the ApplicationWizard component', () => {
  const wrapper = shallow(<CiipApplication query={query} />);
  expect(
    wrapper
      .find('Relay(ApplicationWizard)')
      .first()
      .prop('formJson')
  ).toBe(query.allFormJsons.edges.node);
});
