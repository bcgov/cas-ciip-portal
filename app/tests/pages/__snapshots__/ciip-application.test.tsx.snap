// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It matches the last accepted Snapshot 1`] = `
<DefaultLayout>
  <Relay(ApplicationWizard) />
</DefaultLayout>
`;
